tests/Utilities/MakeWithRandomValues.hpp

```diff
@@ -70,6 +70,53 @@ struct FillWithRandomValuesImpl<Variables<tmpl::list<Tags...>>,
 /// \endcond
 }  // namespace TestHelpers_detail
 
+/// \ingroup TestingFrameworkGroup
+/// \brief A uniform distribution functor for which the behavior can be easily
+/// customized
+///
+/// \details The two constructors take either a low and high bound arguments or
+/// a similar 2-component array , which can be passed in in any type from which
+/// the generated type can be constructed. The default generation function
+/// operates as would be expected for a double
+template <typename T>
+struct uniform_custom_distribution {
+  using result_type = T;
+  std::array<T, 2> bounds;
+  template <typename Bound>
+  uniform_custom_distribution(Bound Low, Bound High) noexcept {
+    bounds = std::array<T, 2>{{T(Low), T(High)}};
+  }
+  template <typename Bound>
+  explicit uniform_custom_distribution(std::array<Bound, 2> bounds) noexcept
+      : bounds{{T(bounds[0]), T(bounds[1])}} {}
+
+  template <class Generator>
+  T operator()(Generator& g) noexcept {
+    return ((T(g()) - Generator::min()) *
+                (gsl::at(bounds, 1) - gsl::at(bounds, 0)) /
+                (Generator::max() - Generator::min()) +
+            gsl::at(bounds, 0));
+  }
+};
+
+/// \ingroup TestingFrameworkGroup
+/// \brief Specialization of the generation function to independently sample the
+/// components of a std:: complex<double
+template <>
+template <class Generator>
+std::complex<double> uniform_custom_distribution<std::complex<double>>::
```

Comment:
Why only complex doubles and not all complex?

Reply:
That constitutes a partial specialization and requires a full class definition, rather than getting to just define the member function. This seemed briefer, but if you want the generality I can make a partial specialization.